programs/steward/src/state/steward_state.rs

```diff
@@ -816,13 +815,16 @@ impl StewardState {
                 .checked_add(stake_rent)
                 .ok_or(StewardError::ArithmeticError)?;
 
-            // Maximum increase amount is the total lamports in the reserve stake account minus 2 * stake_rent, which accounts for reserve rent + transient rent
-            // Saturating_sub because reserve stake may be less than 2 * stake_rent, but needs more than 2 * stake_rent to be able to delegate
-            let reserve_lamports = reserve_lamports.saturating_sub(
-                stake_rent
-                    .checked_mul(2)
-                    .ok_or(StewardError::ArithmeticError)?,
-            );
+            // Maximum increase amount is the total lamports in the reserve stake account minus (num_validators + 1) * stake_rent, which covers rent for all validators plus the transient rent
+            let accounts_needed_reserve_for_rent = validator_list
+                .len()
+                .checked_add(1)
+                .ok_or(StewardError::ArithmeticError)?;
```

Comment:
small optimization: could subtract `progress.count()` here since the ones that have already been run won't need the transient rent

Reply:
Ah, clever

Reply:
Done